### PR TITLE
Added request headers to http tests

### PR DESCRIPTION
**Pull request by dacamp**

### CODE REVIEW

**Comment by dacamp** [comment]
Build is failing because Travis is pinned to `go1.11` up to @aelsabbahy to weigh in on either updating the go version or maintain compatibility:

```diff
diff --git a/system/http.go b/system/http.go
index fc18225..7de0646 100644
--- a/system/http.go
+++ b/system/http.go
@@ -74,7 +74,7 @@ func (u *DefHTTP) setup() error {
        if err != nil {
                return u.err
        }
-       req.Header = u.RequestHeader.Clone()
+       req.Header = u.RequestHeader
        if u.Username != "" || u.Password != "" {
                req.SetBasicAuth(u.Username, u.Password)
        }
```

**Comment by aelsabbahy** [comment]
I'm fine with updating go if it doesn't break anything else.

**Comment by aelsabbahy** [comment]
https://github.com/aelsabbahy/goss/blob/master/.travis.yml#L4

I think this is what you meant to change.

**Comment by dacamp** [comment]
Ahh that's what I was looking for! Want me to backout the changes on `go.mod`?

**Comment by aelsabbahy** [comment]
Can you point me to the docs on the go.mod change. Go has changed dep management systems so many times, I actually haven't read up on go.mod yet.

Guess what I'm saying is, I don't understand the implications of the go.mod change you made. :)

**Comment by aelsabbahy** [comment]
If you get this PR green today I'll probably cut a Goss release with it tomorrow.

**Comment by dacamp** [comment]
This is the[ best doc](https://blog.ksub.org/bytes/2019/09/15/go.mods-go-directive/) I've found to describe the `go` directive in `go.mod`. Looks like it defines the expected language version. Because `header.Clone` was released in `go 1.13` I think it makes sense to set the go version.

**Comment by dacamp** [comment]
Looks like the build if failing in Travis. I've noticed these fail locally prior to making any changes. If you think they're just flaky tests would you mind retrying the build in Travis so I don't have to push a no-op change?

**Comment by aelsabbahy** [comment]
Count: 94, Failed: 0, Skipped: 5
+[[ centos7 == \a\r\c\h ]]
+egrep -q 'Count: 93, Failed: 0, Skipped: 5'


The counts are mismatching since you added a test.

**Comment by aelsabbahy** [comment]
Also, docs need to be updated. I'm doing quick eyeball review on my phone currently.

**Comment by aelsabbahy** [comment]
Thanks for the document on go.mod, explained it very well.

File: docs/manual.md [hunk 1]
```diff
@@ -583,6 +583,8 @@ http:
     allow-insecure: false
     no-follow-redirects: false # Setting this to true will NOT follow redirects
     timeout: 1000
+    request-header: # Optionally you can set request header values
```
Comment (MarkusTeufelberger) [inline docs/manual.md]: Should be `request-headers`, right?

Comment (aelsabbahy) [reply docs/manual.md]: The change was merged already, someone submitted a documentation fix last week.

Next release will fix the header/request-headers inconsistency between the two features.

**Pull request merged**